eotsmanager/cmd/eotsd/daemon/pop.go

```diff
@@ -0,0 +1,319 @@
+	BtcSignBaby string `json:"btcSignBaby"`
+
+	// Btc address is the same as the btc pub key
+	BtcAddress string `json:"btcAddress"`
+	// Babylon address ex.: bbn1f04czxeqprn0s9fe7kdzqyde2e6nqj63dllwsm
+	BabyAddress string `json:"babyAddress"`
+}
+
+func NewExportPopCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "pop-export",
+		Short: "Exports the Proof of Possession by signing over the BABY address with the EOTS private key and the EOTS public key with the BABY private key.",
+		Long: `Parse the address from the BABY keyring, loads the address hash it with
+		sha256 and sign based on the EOTS key associated with the key-name or eots-pk flag.
+		If the both flags are supplied, eots-pk takes priority. Use the generated signature
+		to build a Proof of Possession. For the creation of the BABY signature over the eots pk,
+		it loads the BABY key pair and signs the eots-pk hex with adr036 and exports it.`,
+		RunE: exportPop,
+	}
+
+	f := cmd.Flags()
+
+	f.String(sdkflags.FlagHome, config.DefaultEOTSDir, "EOTS home directory")
+	f.String(keyNameFlag, "", "EOTS key name")
+	f.String(eotsPkFlag, "", "EOTS public key of the finality-provider")
+	f.String(passphraseFlag, "", "EOTS passphrase used to decrypt the keyring")
+	f.String(sdkflags.FlagKeyringBackend, keyring.BackendTest, "EOTS backend of the keyring")
+
+	f.String(flagHomeBaby, "", "BABY home directory")
+	f.String(flagKeyNameBaby, "", "BABY key name")
+	f.String(flagKeyringBackendBaby, keyring.BackendTest, "BABY backend of the keyring")
+
+	return cmd
+}
+
+func exportPop(cmd *cobra.Command, _ []string) error {
+	f := cmd.Flags()
+
+	eotsKeyName, err := f.GetString(keyNameFlag)
+	if err != nil {
+		return err
+	}
+
+	eotsFpPubKeyStr, err := f.GetString(eotsPkFlag)
+	if err != nil {
+		return err
+	}
+
+	eotsPassphrase, err := f.GetString(passphraseFlag)
+	if err != nil {
+		return err
+	}
+
+	eotsKeyringBackend, err := f.GetString(sdkflags.FlagKeyringBackend)
+	if err != nil {
+		return err
+	}
+
+	eotsHomePath, err := getHomePath(cmd)
+	if err != nil {
+		return fmt.Errorf("failed to load home flag: %w", err)
+	}
+
+	babyHomePath, err := getCleanPath(cmd, flagHomeBaby)
+	if err != nil {
+		return fmt.Errorf("failed to load baby home flag: %w", err)
+	}
+
+	babyKeyName, err := f.GetString(flagKeyNameBaby)
+	if err != nil {
+		return err
+	}
+
+	babyKeyringBackend, err := f.GetString(flagKeyringBackendBaby)
+	if err != nil {
+		return err
+	}
+
+	cdc := codec.MakeCodec()
+	babyKeyring, err := keyring.New("baby", babyKeyringBackend, babyHomePath, cmd.InOrStdin(), cdc)
+	if err != nil {
+		return fmt.Errorf("failed to create keyring: %w", err)
+	}
+
+	babyKeyRecord, err := babyKeyring.Key(babyKeyName)
+	if err != nil {
+		return err
+	}
+
+	babyAccAddr, err := babyKeyRecord.GetAddress()
+	if err != nil {
+		return err
+	}
+
+	bbnAddrBz, err := sdk.GetFromBech32(babyAccAddr.String(), "bbn")
+	if err != nil {
+		return err
+	}
+	bbnAddr := sdk.AccAddress(bbnAddrBz)
```

Comment:
seems we already have `bbnAddr` from line 135, why we circle back here?

Reply:
I wasn't sure that the accAddress decoded from line 135 was with our prefix

Reply:
but you are correct, they are the same
Removed unnecessary parse